### PR TITLE
Remove hack obsoleted by PEP 655

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
Huh, it seems that while pytype supports `NotRequired`, it doesn't support `Required`. I'll invert the TypeDict and try again.

**Comment by srittau** [comment]
google/pytype#1535

File: stdlib/logging/config.pyi [hunk 1]
```diff
@@ -50,17 +50,15 @@ _FilterConfiguration: TypeAlias = _FilterConfigurationTypedDict | dict[str, Any]
...
-class _DictConfigArgs(_OptionalDictConfigArgs, TypedDict):
+class _DictConfigArgs(TypedDict):
     version: Literal[1]
+    formatters: NotRequired[dict[str, _FormatterConfiguration]]
+    filters: NotRequired[dict[str, _FilterConfiguration]]
+    handlers: NotRequired[dict[str, _HandlerConfiguration]]
+    loggers: NotRequired[dict[str, _LoggerConfiguration]]
+    root: NotRequired[_RootLoggerConfiguration]
+    incremental: NotRequired[bool]
+    disable_existing_loggers: NotRequired[bool]
```
Comment (AlexWaygood) [inline stdlib/logging/config.pyi:61]: Ugh... maybe we should just wait for pytype to fix the `Required` bug? I agree the existing hack is ugly, but this seems pretty ugly too :(

Comment (srittau) [reply stdlib/logging/config.pyi:61]: While this is also ugly, I still prefer it to the old hack, which is a bit hard to understand. That said, maybe the pytype problem gets fixed quickly.

Comment (rchen152) [reply stdlib/logging/config.pyi:61]: Yeah, this should be a quick fix in pytype =) I'll try to get a release out in the next day or two.

Comment (rchen152) [reply stdlib/logging/config.pyi]: Pytype release 2023.11.29 contains the fix.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**